src/topology/subset_properties.lean

```diff
@@ -382,10 +381,8 @@ end tychonoff
 instance quot.compact_space {r : α → α → Prop} [compact_space α] :
   compact_space (quot r) :=
 ⟨begin
-   have : quot.mk r '' univ = univ,
-     by rw [image_univ, range_iff_surjective]; exact quot.exists_rep,
-   rw ←this,
-   exact compact_image compact_univ continuous_quot_mk
+  convert ← compact_univ.image continuous_quot_mk; [skip, assumption],
```

Comment:
You are going a dangerous Mario-like direction, making proofs shorter even if it makes them less readable :) Here, it is not even shorter as there were four lines before and four lines after if you write one command by line. Joke aside, keeping some `have` statements helps readability a lot.

Reply:
OK, I'll reverse this.

Reply:
BTW, why does `convert` fail to get an instance from assumptions here?

Reply:
I agree that `have`/`obtain` are important in long proofs, and I try to use them. However in trivial proofs like this one too many `have`s hide the (trivial) "main" step from the reader. I rewrote it adding a lemma to `set/basic` and rewriting on this lemma before using `exact`.